src/metldata/builtin_transformations/count_content_values/data_transform.py

```diff
@@ -0,0 +1,132 @@
+    get_directly_referenced_class,
+)
+from metldata.builtin_transformations.count_content_values.instruction import (
+    CountContentValueInstruction,
+)
+from metldata.transform.exceptions import (
+    EvitableTransformationError,
+)
+
+
+class TransformationContext:
+    """Context manager for a data transformation process, wrapping the Datapack to be
+    transformed and transformation instructions per class.
+    """
+
+    def __init__(
+        self,
+        data: DataPack,
+        instructions_by_class: dict[str, list[CountContentValueInstruction]],
+    ):
+        self.data = data.model_copy(deep=True)
+        self.instructions_by_class = instructions_by_class
+
+    def get_class_resources(self, class_name: str):
+        """Extract the resources of a given class from a Datapack."""
+        resources = self.data.resources.get(class_name)
+        if not resources:
+            raise EvitableTransformationError()
+        return resources
+
+
+def count_content(
+    *,
+    data: DataPack,
+    instructions_by_class: dict[str, list[CountContentValueInstruction]],
+) -> DataPack:
+    """Transforms the data."""
+    context = TransformationContext(data, instructions_by_class)
+    for class_name, instructions in context.instructions_by_class.items():
+        resources = context.get_class_resources(class_name)
+
+        for instruction in instructions:
+            relation_path = instruction.source.relation_path
+            referenced_class = get_directly_referenced_class(relation_path)
+
+            # Only one element is expected in the path
+            relation_name = relation_path.elements[0].property
+            # content_resources keeping the values to be count are obtained from
+            # the class that is referenced by the transformation target class through
+            # relation_name.
+            content_resources = context.get_class_resources(referenced_class)
+
+            transform_resources(
+                resources, relation_name, content_resources, instruction
+            )
+
+    return context.data
+
+
+def transform_resources(
+    resources,
+    relation_name,
+    content_resources,
+    instruction: CountContentValueInstruction,
+):
+    """Transform resources"""
+    for resource in resources.values():
+        content = resource.content
+        related_to = resource.relations.get(relation_name)
+        count_values = get_count_values(related_to, content_resources, instruction)
+
+        target_object = get_target_object(content, instruction)
+        target_object[instruction.target_content.property_name] = dict(
+            Counter(count_values)
+        )
+
+
+def get_count_values(
+    resource_relations, content_resources, instruction: CountContentValueInstruction
+):
+    """Get values that is to be counted."""
+    if not resource_relations:
+        raise EvitableTransformationError()
+
+    try:
+        count_values = [
```

Comment:
```suggestion
        value_counts = [
```

Reply:
But the actual count happens in `transform_resources`. This extracts values that we count later on. `count_values` alludes that these values will be used in the count.

Reply:
I changed it to `content_values` - might explain it better.